### PR TITLE
fix query string syntax doc when OR operator is missed

### PR DESCRIPTION
**Pull request by fbaligand**
@jimczi 
I complete here your PR #28798 to fix issue #28719

### CODE REVIEW

File: docs/reference/query-dsl/query-string-syntax.asciidoc [hunk 1]
```diff
@@ -24,7 +24,7 @@ search terms, but it is possible to specify other fields in the query syntax:
     status:active
 
 * where the `title` field contains `quick` or `brown`.
-  If you omit the OR operator the default operator will be used
+  If you omit the OR operator, the default operator will be used (except if `title` field is typed `keyword`)
```
Comment (jimczi) [inline docs/reference/query-dsl/query-string-syntax.asciidoc]: I think we should remove this sentence completely. We don't split on whitespace anymore so we shouldn't assume that `quick` and `brown` are going to be separated. It depends only on the tokenizer that is use for the field. We've added a warning section at the beginning of the docs regarding `keyword` field so I think it's enough.

File: docs/reference/query-dsl/query-string-syntax.asciidoc [hunk 2]
```diff
@@ -36,7 +36,7 @@ search terms, but it is possible to specify other fields in the query syntax:
 * where any of the fields `book.title`, `book.content` or `book.date` contains
   `quick` or `brown` (note how we need to escape the `*` with a backslash):
 
-    book.\*:(quick brown)
+    book.\*:(quick OR brown)
```
Comment (jimczi) [inline docs/reference/query-dsl/query-string-syntax.asciidoc]: +1 to add an explicit `OR`

**Review by jimczi — CHANGES_REQUESTED** [review]
I left some comments, I think we should just remove the sentence that you changed.

**Comment by fbaligand** [comment]
@jimczi 
I agree with you.
So I removed the sentence, and the associated example `title:(quick brown)`

**Comment by fbaligand** [comment]
@jimczi 
Could you review my last change and tell if it answers your requested changes ?

**Comment by fbaligand** [comment]
Hi @jimczi
Could you review my last commit ?

**Review by jimczi — APPROVED** [review]
Thanks @fbaligand

**Comment by fbaligand** [comment]
Thanks @jimczi for the merge !

**Pull request merged**